File: apps/telegram_mvp_kmp/shared/src/commonMain/kotlin/com/hypheng/telegrammvpkmp/HomeTab.kt
package com.hypheng.telegrammvpkmp

enum class HomeTab(val id: String) {
    Chats("chats"),
    Contacts("contacts"),
    Settings("settings");

    companion object {
        fun fromId(id: String): HomeTab = entries.firstOrNull { it.id == id } ?: Chats
    }
}

File: apps/telegram_mvp_flutter/lib/src/features/home/home_tab.dart
enum HomeTab { chats, contacts, settings }

extension HomeTabIndex on HomeTab {
  int get index => switch (this) {
    HomeTab.chats => 0,
    HomeTab.contacts => 1,
    HomeTab.settings => 2,
  };

  String get routePath => switch (this) {
    HomeTab.chats => '/home/chats',
    HomeTab.contacts => '/home/contacts',
    HomeTab.settings => '/home/settings',
  };
}
